cmd/svcat/command/classfiltered.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package command
+
+import (
+	"github.com/spf13/cobra"
+)
+
+// HasClassFlag represents a command that supports --class.
+type HasClassFlag interface {
+	// ApplyClassFlag validates and persists the class related flag.
+	//   --class
+	ApplyClassFlag(*cobra.Command) error
+}
+
+// ClassFilteredCommand adds support to a command for the --class flag.
+type ClassFilteredCommand struct {
```

Comment:
FYI, Morgan and I decided on a slightly different naming convention so that it's not so stuttered/verbose

https://github.com/kubernetes-incubator/service-catalog/blob/0eb083fd800e3d0bc658e5ccc285951c2c3e33a1/cmd/svcat/instance/get_cmd.go#L37

Basically since the package is already called `command`, we don't need the `*Command` suffix on the type and the constructor below.

I'm cool with that being in a follow-up issue/PR though. I realize we changed that out from under you and I apologize for the churn! 😊